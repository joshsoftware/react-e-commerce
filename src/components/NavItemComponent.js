import React from 'react';
import NavLinkComponent from './NavLinkComponent';
import { NavItem } from 'reactstrap';
import PropTypes from 'prop-types';

const NavItemComponent = (props) => {
  const { url, lable, logo } = props;
  return (
    <NavItem>
      <NavLinkComponent url={url} lable={lable} logo={logo} />
    </NavItem>
  );
};

export default NavItemComponent;

NavItemComponent.propTypes = {
  url: PropTypes.string.isRequired,
  lable: PropTypes.string,
  logo: PropTypes.string
};
