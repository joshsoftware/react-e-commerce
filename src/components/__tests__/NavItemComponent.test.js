import NavItemComponent from '../NavItemComponent';
import { render } from '@testing-library/react';
import React from 'react';

describe('NavItem Component', () => {
  it('Must return Nav Item', () => {
    const url = '/';
    const lable = 'Lable';
    const logo =
      'https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_1280.png';
    const { asFragment } = render(<NavItemComponent url={url} lable={lable} logo={logo} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
