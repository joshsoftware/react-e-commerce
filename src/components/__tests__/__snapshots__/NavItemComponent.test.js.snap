// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem Component Must return Nav Item 1`] = `
<DocumentFragment>
  <li
    class="nav-item"
  >
    <a
      class="nav-link"
      href="/"
    >
      <img
        alt="img"
        class="rounded-circle"
        src="https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_1280.png"
        style="width: 70px;"
      />
    </a>
  </li>
</DocumentFragment>
`;
